### ISSUE TITLE
Import from KOReader Vocab Builder cannot find the files it needs

### ISSUE THREAD

**Issue by billyc** [FreeLanguageTools/vocabsieve#70]
**Describe the bug**

Import from KOReader Vocab Builder fails upon selecting the folder with all of my KOReader files. The error dialog appears: "Check if you've picked the right directory. It should be a folder containing both all your books and KOReader settings."

The folder I select is the folder containing all of the KOReader data files synced from my Android KOReader installation. This includes a "books" folder, which includes `.epub` files and a subfolder for each book with the appropriate `.sdr` folder within it. The folder also contains the KOReader `settings` folder which contains the `vocabulary_builder.sqlite3` database.

**To Reproduce**

1. Copy or sync all files from KOReader device to a desktop folder, in this case macOS
2. Click on Import > KOReader Vocab Builder
3. Select the folder containing the synced KOReader files
4. See error: the error dialog attached below appears.

**Expected behavior**

The import should work since all books, `.sdr folders, and the vocab builder sqlite3 file are all present in the synced folder

**Screenshots**

![image](https://github.com/FreeLanguageTools/vocabsieve/assets/1427526/0a5ce76e-e812-4db5-9b45-ab916d3f963c)

**Desktop (please complete the following information):**
- OS: macOS 12.6.4 "Monterey"
- Vocabsieve version v0.10.1, installed from GitHub Releases page .dmg file

**Additional context**

KOReader has a "cloud sync" option which syncs just the sqlite3 file with the vocab builder data, but apparently that is not sufficient. The error dialog says the books (and I assume the .sdr folders) must also be present. So, I have copied the entire KOReader folder instead using "SyncThing", and I have verified that all of the files are present in the synced folder on my Mac desktop.

Happy to provide more details or do testing/triage.

**Comment by 1over137** [comment]
Why would there be .srt files for books? Though having extra files shouldn't affect it in any way. The vocab builder uses the metadata that resides in the corresponding .sdr folders. Can you show the folder structure in detail?

**Comment by billyc** [comment]
@1over137 I meant the `.sdr` folder, not .srt. My apologies, I've updated the original bug description. Just a typo.

Here is a screenshot of the synced folder. You can see that the entire koreader tree is there, and I've drilled down into koreader/books/Çinar Ağacı.sdr/ so you can see .epub and the .sdr folder  contents:

<img width="926" alt="image" src="https://github.com/FreeLanguageTools/vocabsieve/assets/1427526/40c36680-20a0-4e04-a118-cca80180194b">

---
*and here is the content of the settings folder:*

<img width="697" alt="image" src="https://github.com/FreeLanguageTools/vocabsieve/assets/1427526/c2b13dd8-4fee-4704-bf58-db0b4acb785e">

---

Here is a .zip file with the entire contents of the synced koreader folder, in case that helps us debug what's going on...

https://tubcloud.tu-berlin.de/s/RJwXk5T8XXLYWkk

**Comment by billyc** [comment]
I found the problem - it is related to books with long language codes such as `en-US`; any book with its language set to the correct main language but with an additional region code fails to get parsed.

In my case I set my target language in VocabSieve to Turkish which is language code "tr", but all of my ePubs have their language code set to "tr-TR" and thus the import fails.

The PR #73 linked above fixes this.
